enum.c

```diff
@@ -93,6 +110,20 @@ grep_iter_i(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
     return Qnil;
 }
 
+static VALUE
+enum_grep0(VALUE obj, VALUE pat, VALUE test)
+{
+    VALUE ary = rb_ary_new();
+    struct MEMO *memo = MEMO_NEW(pat, ary, test);
+    rb_block_call_func_t fn = rb_block_given_p() ? grep_iter_i :
```

Comment:
Can we switch to if/else if/else instead of nested ternary operator use here?

Reply:
Sure 😅 I had written it that way first, I should have kept it this way 😆